### PR TITLE
Bug 13490813 - Create Prerelease Longitudinal

### PR DESCRIPTION
**Pull request by fbertsch**
This is similar to my previous PR but without the per-process stuff. As such we won't be able to actually create this dataset yet, but the ability is there. I just wanted to use the scalar and histogram parts I created in here for some main_summary work.

Also, now the options are: an opt-out only dataset, or an opt-out and opt-in dataset. No more of the confusing "just opt-in" parts.

### CODE REVIEW

**Comment by fbertsch** [comment]
@harterrt will you be able to get to this r anytime soon? If not we can pass it on to someone else.

**Comment by harterrt** [comment]
Hey, Frank. Sorry for the radio silence :(. 

This is on my radar, but I probably will not be able to review until Friday. I see now that this is blocking some of your work on main_summary. Is that time-line reasonable or should I hand off this review?

**Comment by fbertsch** [comment]
@harterrt Friday works. Thanks for letting me know :)

File: src/test/scala/com/mozilla/telemetry/histograms/HistogramsTest.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  val fixture = {
+    new {
+      val histograms = new HistogramsClass{
+        override protected val getURL = (a: String, b: String) => Source.fromFile("src/test/resources/ShortHistograms.json")
+      }
+
+      private val optStatuses = List(false, true)
+
+      val names = optStatuses map {
+        case (optStatus) =>
```
Comment (harterrt) [inline src/test/scala/com/mozilla/telemetry/histograms/HistogramsTest.scala]: Can you drop the case statement here?

File: src/test/scala/com/mozilla/telemetry/histograms/HistogramsTest.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            optStatus -> histograms.definitions(optStatus).map{
+              case (k, value) =>
```
Comment (harterrt) [inline src/test/scala/com/mozilla/telemetry/histograms/HistogramsTest.scala]: Is the case statement necessary here?

File: src/test/scala/com/mozilla/telemetry/LongitudinalTest.scala [hunk 3]
```diff
@@ -197,34 +224,57 @@ class LongitudinalTest extends FlatSpec with Matchers with PrivateMethodTester {
...
       sc.setLogLevel("WARN")
+
       private val sqlContext = new SQLContext(sc)
-      val rows = sqlContext.read.load(filename).collect()
-      val row =  rows(0)
+
+      // Opt-out only schema
+      private var includeOptin = false
+      private var histogramDefs = histograms.definitions(includeOptin)
+      private var scalarDefs = scalars.definitions(includeOptin)
```
Comment (harterrt) [inline src/test/scala/com/mozilla/telemetry/LongitudinalTest.scala]: This is personal opinion so these suggestions are totally optional :).

These three lines may be more readable if you used named arguments like `definitions(includeOptin = false)`. Please also consider changing the `var`s to `val`s. Same with the next three statements.

File: src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala [hunk 4]
```diff
@@ -684,21 +713,25 @@ object LongitudinalView {
     root.set(avroField, fieldValues.asJava)
   }
 
+  def getStandardHistogramSchema(schema: Schema): Schema = {
```
Comment (fbertsch) [inline src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala]: Hate repeating that random string. This function makes it more clear what we're doing.

File: src/test/scala/com/mozilla/telemetry/histograms/HistogramsTest.scala (hunk 2 above)
Comment (fbertsch) [inline src/test/scala/com/mozilla/telemetry/histograms/HistogramsTest.scala]: Good practice. See http://docs.scala-lang.org/cheatsheets/#pattern_matching

File: src/test/scala/com/mozilla/telemetry/LongitudinalTest.scala (hunk 3 above)
Comment (fbertsch) [inline src/test/scala/com/mozilla/telemetry/LongitudinalTest.scala]: It was good advice :)

File: src/test/scala/com/mozilla/telemetry/histograms/HistogramsTest.scala (hunk 2 above)
Comment (harterrt) [inline src/test/scala/com/mozilla/telemetry/histograms/HistogramsTest.scala]: Learn something new every day! Thanks.

File: src/test/scala/com/mozilla/telemetry/scalars/ScalarsTest.scala [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  val fixture = {
+    new {
+      val scalars = new ScalarsClass{
+        override protected val getURL = (a: String, b: String) => Source.fromFile("src/test/resources/Scalars.yaml")
+      }
+
+      private val optStatuses = List(true, false)
+
+      val names = optStatuses map {
+        case (optStatus) =>
```
Comment (harterrt) [inline src/test/scala/com/mozilla/telemetry/scalars/ScalarsTest.scala]: Please remove this case statement as well.

**Comment by fbertsch** [comment]
@harterrt fixed - RFAL

**Pull request merged**